doc/1_applications.dox

```diff
@@ -0,0 +1,160 @@
+/**
+
+@page applications Sample Applications
+
+@tableofcontents
+
+@section application_introduction Introduction
+
+The easiest way to get started for many people is with samples. We've provided
+the source for some applications. Some of these applications you can run
+directly from the command line. While others will require a device.
+
+@section application_folders Directory Organization
+
+The sample applications typically have the following folder structure:
+
+    targets/
+    config.mk
+    Makefile
+    subdirs
+
+The code for an application typcially consists of two parts: shared and
+target-specific. The shared code will be at the top folder for the application.
+While the target-specific information will be under a target folder. As an
+example, the blink_raw application has shared code here:
+
+    applications/blink_raw/
+        main.cxx
+        targets/freertos.armv7m.mbed-1768/
+            hw_init.cxx
+
+The idea is that you can have target-specific (platform-specific) code that
+allows customizing the shared code for different platforms.
+
+@subsection application_build Building a Sample
+
+Generally when you build a sample, you'll want to do so from a single target
+directory, as that will build faster and only build what you need. Here we're
+going to build the Linux versions so you can run them without needing a
+specific board.
+
+@section grid_connect GridConnect Protocol
+
+Some of the sample applications use a protocol referred to here as the
+GridConnect protocol. This is an ASCII protocol that is used by USB-to-CAN
+interfaces sold by [GridConnect](http://gridconnect.com/).
+
+A normal CAN message has the following format:
+
+:X <_Identifier_> <N> <_Data-0_><_Data-1_>...<_Data-7_>;
+
+These messages are sent within a single network (such as a CAN network or over
+TCP) as binary. But if you're using one of the GridConnect devices, their
+I/O format is text. Here is an actual example:
+
+    :X19545415N0502010202000000;
+
+The ID is 19545415. LCC uses 29-bit CAN identifiers. The data bytes are encoded
+as HEX.
+
+You can also have messages that contains no extra data. For example:
+
+    :X17050415N;
+
+A couple of the sample applications output this format to the console in order
+to display the traffic. But, again, the actual CAN/TCP messages will be
+binary messages.
```

Comment:
Actually, while over CAN, they are binary, over TCP (when using GridConnect), they are encoded as GridConnect ASCII.  There is work being done on a more efficient (and powerful) binary TCP protocol and a draft OpenLCB standard for it.